3251 protect api with password

entantoencuanto opened the pull request:
Closes #3251


## :v: What does this PR do?

* Adds basic auth to each action of API if site is protected with password.
* This filter is skipped if the user provides a valid user or admin authentication token (the user must be active or admin must have permissions on site)
* Adds tests to site navigation and API

## :mag: How should this be manually tested?

Nothing should change in gobierto data, investments and plans

## :shipit: Does this PR changes any configuration file?

No

(Changes in these files might need to update the role in Ansible)

## :book: Does this PR require updating the documentation?

Suggestion sent to gobierto.readme.io

entantoencuanto committed:
Apply basic authentication on API requests on protected sites

entantoencuanto committed:
Provide basic auth token in gobierto footer if required

entantoencuanto committed:
Change getToken to include basic_auth_token if no other auth token is present

entantoencuanto committed:
Skip protected site authentication if user provides his own auth token

entantoencuanto committed:
Include basic_auth_token in gobierto plans factories header

ferblape reviewed (changes requested):
I miss:

- a clarification in the documentation (both in how the module works and the api docs)
- tests

entantoencuanto committed:
Add tests for navigation and API with password protected sites

entantoencuanto committed:
Skip site password protection in data API if authorized admin token is provided

entantoencuanto committed:
Change API to give access to password protected sites with valid tokens

If the request includes an authorization token of a valid user or admin
the basic authorization checking is skipped

entantoencuanto committed:
Add test for investments API access when site is protected with password
